File: Ruby/test/applitools_website_test.rb
require 'test/unit'
require 'eyes_selenium'

class MyTest < Test::Unit::TestCase

  def setup
    @eyes = Applitools::Eyes.new
    # This is your api key, make sure you use it in all your tests.
    @eyes.api_key = APPLITOOLS_APIKEY

    # Get a selenium web driver object.
    @driver = Selenium::WebDriver.for :firefox
  end

  # Called after every test method runs. Can be used to tear
  # down fixture information.

  def teardown
    @eyes.abort_if_not_closed
  end

  # Fake test
  def test
    @eyes.test(app_name: 'Applitools website', test_name: 'Example test',
               viewport_size: {width: 900, height: 600}, driver: @driver) do |wrapped_driver|
      wrapped_driver.get 'http://www.applitools.com'
      # Visual validation point #1
      @eyes.check_window('Main Page')
      wrapped_driver.find_element(:css, ".read_more").click
      # Visual validation point #2
      @eyes.check_window('Features Page')
    end
  end
end

File: Java/src/test/java/ApplitoolsWebsiteTest.java

import com.applitools.eyes.RectangleSize;
import com.applitools.eyes.selenium.Eyes;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class ApplitoolsWebsiteTest {
    private WebDriver driver_;
    private Eyes eyes_;

    @Before
    public void setup() {
        driver_ = new FirefoxDriver();
        eyes_ = new Eyes();
    }

    @Test
    public void test() {
        driver_.get("http://applitools.com/helloworld");

        // Start visual testing with browser viewport set to 1024x768.
        // Make sure to use the returned driver from this point on.
        driver_ = eyes_.open(driver_, "Applitools website", "Test Applitools homepage", new RectangleSize(900, 700));

        // Visual validation point #1
        eyes_.checkWindow("Home Page");

        driver_.findElement(By.cssSelector("button")).click();

        // Visual validation point #2
        eyes_.checkWindow("Features page");

        // End visual testing. Validate visual correctness.
        eyes_.close();
    }

    @After
    public void tearDown() {
        eyes_.abortIfNotClosed();
    }
}
